code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,448 @@
+	if(health > 1)
+		health = 1
+		..()
+	health = 0
+
+/mob/living/simple_animal/hostile/swarmer/CanPass(atom/movable/O)
+	if(istype(O, /obj/item/projectile/beam/disabler))//Allows for swarmers to fight as a group without wasting their shots hitting each other
+		return 1
+	..()
+
+////CTRL CLICK FOR SWARMERS AND SWARMER_ACT()'S////
+/mob/living/simple_animal/hostile/swarmer/CtrlClickOn(atom/A)
+	face_atom(A)
+	if(!isturf(loc))
+		return
+	if(next_move > world.time)
+		return
+	if(!A.Adjacent(src))
+		return
+	A.swarmer_act(src)
+	return
+
+/atom/proc/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S.DisIntegrate(src)
+
+/obj/item/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S.Integrate(src)
+
+/obj/item/weapon/gun/swarmer_act()//Stops you from eating the entire armory
+	return
+
+/turf/simulated/floor/swarmer_act()
+	return
+
+/obj/structure/reagent_dispensers/fueltank/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>Destroying this object would cause a chain reaction. Aborting.</span>"
+
+/obj/machinery/portable_atmospherics/canister/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>An inhospitable area may be created as a result of destroying this object. Aborting.</span>"
+
+/obj/structure/cable/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>Disrupting the power grid would bring no benefit to us. Aborting.</span>"
+
+/obj/machinery/power/apc/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>Disrupting the power grid would bring no benefit to us. Aborting.</span>"
+
+/obj/machinery/power/smes/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>Disrupting the power grid would bring no benefit to us. Aborting.</span>"
+
+/turf/simulated/wall/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	if(locate(/turf/space) in range(1, src))
+		S << "<span class='warning'>Destroying this object has the potential to cause a hull breach. Aborting.</span>"
+		return
+	..()
+
+/obj/structure/window/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	if(locate(/turf/space) in range(1, src))
+		S << "<span class='warning'>Destroying this object has the potential to cause a hull breach. Aborting.</span>"
+		return
+	..()
+
+/obj/machinery/power/emitter/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)//Stop trying to release the singularity you fuck
+	S << "<span class='warning'>This object is too risky to destroy. Aborting.</span>"
+
+/obj/item/stack/cable_coil/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)//Wiring would be too effective as a resource
+	S << "<span class='warning'>This object does not contain enough materials to work with.</span>"
+
+/obj/machinery/porta_turret/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	S << "<span class='warning'>This machine has been warded against us!</span>"
```

Comment:
How about "Attempting to destroy this machine would result in our destruction" given that it is a turret, it might shoot them or something.